pkg/writers/buffered_file_writer/bufferedfilewriter.go

```diff
@@ -8,29 +8,60 @@ import (
 	"io"
 	"os"
 	"sync"
+	"time"
 
 	"github.com/trufflesecurity/trufflehog/v3/pkg/cleantemp"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+type bufferPoolMetrics struct{}
+
+func (b *bufferPoolMetrics) recordGrowth(growthAmount int) {
+	growCount.Inc()
+	growAmount.Add(float64(growthAmount))
+}
+
+func (b *bufferPoolMetrics) recordShrink(amount int) {
+	shrinkCount.Inc()
+	shrinkAmount.Add(float64(amount))
+}
+
+func (b *bufferPoolMetrics) recordCheckoutDuration(duration time.Duration) {
+	checkoutDuration.Observe(float64(duration.Microseconds()))
+	checkoutCount.Inc()
+	checkoutDurationTotal.Add(float64(duration.Microseconds()))
+}
+
+func (b *bufferPoolMetrics) recordBufferRetrival() {
+	activeBufferCount.Inc()
+	bufferCount.Inc()
+}
+
+func (b *bufferPoolMetrics) recordBufferReturn(bufCap, bufLen int64) {
+	activeBufferCount.Dec()
+	totalBufferSize.Add(float64(bufCap))
+	totalBufferLength.Add(float64(bufLen))
+}
+
 type bufPoolOpt func(pool *bufferPool)
 
 type bufferPool struct {
 	bufferSize uint32
 	*sync.Pool
+
+	metrics *bufferPoolMetrics
 }
 
-const defaultBufferSize = 2 << 10 // 2KB
+const defaultBufferSize = 2 << 12 // 4KB
```

Comment:
**Nit:** `2 << 12` is 8KB. `1 << 12` is 4KB. I'd recommend using a 1 since it's easy to get confused between shifting and `2**12`.